migrations/000032_app_versions_table.up.sql

```diff
@@ -0,0 +1,489 @@
+        WHERE app_documentation.app_id = app_versions.app_id
+    );
+ALTER TABLE ONLY app_documentation
+    ALTER COLUMN app_version_id SET NOT NULL;
+ALTER TABLE ONLY app_documentation
+    DROP COLUMN IF EXISTS app_id;
+
+--
+-- `app_documentation` table primary key.
+--
+ALTER TABLE app_documentation
+    ADD CONSTRAINT app_documentation_pkey
+    PRIMARY KEY (app_version_id);
+
+--
+-- Foreign Key for `app_version_id` column in `app_documentation` table.
+--
+ALTER TABLE ONLY app_documentation
+    ADD CONSTRAINT app_documentation_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+--
+-- Replace the `app_id` column in the `app_references` table
+-- with a new `app_version_id` column.
+--
+ALTER TABLE ONLY app_references
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE app_references
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE app_references.app_id = app_versions.app_id
+    );
+ALTER TABLE ONLY app_references
+    ALTER COLUMN app_version_id SET NOT NULL;
+ALTER TABLE ONLY app_references
+    DROP COLUMN IF EXISTS app_id;
+
+--
+-- Foreign Key for `app_version_id` column in `app_references` table.
+--
+ALTER TABLE ONLY app_references
+    ADD CONSTRAINT app_references_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+CREATE INDEX IF NOT EXISTS app_references_app_version_id ON app_references(app_version_id);
+
+--
+-- Replace the `apps_id` column in the `apps_htcondor_extra` table
+-- with a new `app_version_id` column.
+--
+ALTER TABLE ONLY apps_htcondor_extra
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE apps_htcondor_extra
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE apps_htcondor_extra.apps_id = app_versions.app_id
+    );
+ALTER TABLE ONLY apps_htcondor_extra
+    ALTER COLUMN app_version_id SET NOT NULL;
+ALTER TABLE ONLY apps_htcondor_extra
+    DROP COLUMN IF EXISTS apps_id;
+
+--
+-- `apps_htcondor_extra` table primary key.
+--
+ALTER TABLE apps_htcondor_extra
+    ADD CONSTRAINT apps_htcondor_extra_pkey
+    PRIMARY KEY (app_version_id);
+
+--
+-- Foreign Key for `app_version_id` column in `apps_htcondor_extra` table.
+--
+ALTER TABLE ONLY apps_htcondor_extra
+    ADD CONSTRAINT apps_htcondor_extra_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+--
+-- Drop duplicate columns from the `apps` table
+--
+ALTER TABLE ONLY apps
+    DROP COLUMN IF EXISTS deleted;
+ALTER TABLE ONLY apps
+    DROP COLUMN IF EXISTS disabled;
+ALTER TABLE ONLY apps
+    DROP COLUMN IF EXISTS integration_data_id;
+ALTER TABLE ONLY apps
+    DROP COLUMN IF EXISTS integration_date;
+ALTER TABLE ONLY apps
+    DROP COLUMN IF EXISTS edited_date;
+
+--
+-- A view containing the top-level information needed for app listings.
+--
+CREATE VIEW app_listing AS
```

Comment:
This view continues to scare me and I wonder how it affects performance, but I'm not sure that's an actionable comment, heh